Replace org.codehaus.jackson with FasterXML/jackson

This issue is the same as https://issues.apache.org/jira/browse/HELIX-747 (it's a bit confusing which issue list should be used for this project as the documentation on https://cwiki.apache.org/confluence/display/HELIX/Contributor+Workflow doesn't seem to be up to date):

The current json lib Helix uses is out of date. We should consider replacing it with a well-maintained lib.

FasterXML/jackson is compatible with the current lib we used. So it could be a good candidate.

The old version of Jackson that Helix currently uses contains multiple CVEs:
CVE-2017-17485, CVE-2017-7525, CVE-2017-15095

I forked this repository and made a commit that replaces the old Jackson with the newer FasterXML/Jackson:
https://github.com/brunnsbe/helix/commit/ca36d5e51c00e0278fa8526581a9ed9081e619c1
I however have problems running all the tests throws an OutOfMemoryException:
```
...
[ERROR]   TestZkHelixPropertyStore.testSet » OutOfMemory unable to create new native thr...
[ERROR]   TestZkHelixPropertyStore.testSetInvalidPath » OutOfMemory unable to create new...
[ERROR]   TestZkHelixPropertyStore.testZkClientMonitor » OutOfMemory unable to create ne...
[ERROR]   TestZkHelixPropertyStore.testZkTriggeredCallback » OutOfMemory unable to creat...
[ERROR]   TestZkManagerWithAutoFallbackStore.testBasic » OutOfMemory unable to create ne...
[INFO]
[ERROR] Tests run: 866, Failures: 177, Errors: 0, Skipped: 103
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Helix 0.9.1-SNAPSHOT:
[INFO]
[INFO] Apache Helix ....................................... SUCCESS [  2.061 s]
[INFO] Apache Helix :: Core ............................... FAILURE [  01:00 h]
[INFO] Apache Helix :: Admin Webapp ....................... SKIPPED
[INFO] Apache Helix :: Restful Interface .................. SKIPPED
[INFO] Apache Helix :: HelixAgent ......................... SKIPPED
[INFO] Apache Helix :: Recipes ............................ SKIPPED
[INFO] Apache Helix :: Recipes :: Rabbitmq Consumer Group . SKIPPED
[INFO] Apache Helix :: Recipes :: Rsync Replicated File Store SKIPPED
[INFO] Apache Helix :: Recipes :: distributed lock manager  SKIPPED
[INFO] Apache Helix :: Recipes :: distributed task execution SKIPPED
[INFO] Apache Helix :: Recipes :: service discovery ....... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:00 h
[INFO] Finished at: 2019-08-13T09:23:19+03:00
[INFO] ------------------------------------------------------------------------
...
```

--- a/helix-common/pom.xml
+++ b/helix-common/pom.xml
@@ -31,7 +31,6 @@
 
   <properties>
     <osgi.import>
-      org.codehaus.jackson*,
       org.slf4j*;version="[1.6,2)",
       *
     </osgi.import>

--- a/helix-core/pom.xml
+++ b/helix-core/pom.xml
@@ -38,7 +38,6 @@
       org.apache.commons.io*;version="[1.4,2)",
       org.apache.commons.math*;version="[2.1,4)",
       org.apache.jute*;resolution:=optional,
-      org.codehaus.jackson*;version="[1.8,2)",
       org.slf4j*;version="[1.6,2)",
       *
     </osgi.import>

--- a/helix-rest/pom.xml
+++ b/helix-rest/pom.xml
@@ -32,7 +32,6 @@
   <properties>
     <osgi.import>
       org.apache.helix*,
-      org.codehaus.jackson*,
       org.apache.commons.cli*,
       org.restlet*,
       org.slf4j*;version="[1.6,2)",

